template/app/Main.hs

```diff
@@ -0,0 +1,62 @@
+-- SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io>
+-- SPDX-License-Identifier: Unlicense
+
+{-# LANGUAGE ApplicativeDo #-}
+module Main
+  ( main
+  ) where
+
+import Universum
+
+import qualified Data.Map as Map
+import Data.Version (showVersion)
+import Lorentz (DGitRevision, GitRepoSettings(..), mkDGitRevision)
+import Lorentz.ContractRegistry
+import Main.Utf8 (withUtf8)
+import qualified Options.Applicative as Opt
+import Options.Applicative.Help.Pretty (Doc, linebreak)
+import Paths_templateDAO (version)
+
+import qualified Lorentz.Contracts.BaseDAO as DAO
+import qualified Lorentz.Contracts.TemplateDAO as TemplateDAO
+
+programInfo :: DGitRevision -> Opt.ParserInfo CmdLnArgs
+programInfo gitRev = Opt.info (Opt.helper <*> versionOption <*> argParser contracts gitRev) $
+  mconcat
+  [ Opt.fullDesc
+  , Opt.progDesc "MyDAO contracts registry"
+  , Opt.header "MyDAO contracts for Michelson"
+  , Opt.footerDoc usageDoc
+  ]
+  where
+    versionOption = Opt.infoOption ("MyDAO-" <> showVersion version)
+      (Opt.long "version" <> Opt.help "Show version.")
+
+repoSettings :: GitRepoSettings
+repoSettings = GitRepoSettings $ \commit ->
+  "https://github.com/tqtezos/baseDAO/tree/" <> commit
+  -- TODO: update this to refer to your repository, the link will appear in
```

Comment:
Maybe let's provide `TODO` for each action that the user should take? Then in "How to write your own DAOs" we can tell the user to go through all TODOs and do what's written there.

Reply:
Hm, I can try. Probably it should look as follows:
1. `TemplateDAO.hs` have to be read carefully and updated entirely; so I'll leave a point in the `README` saying this about `TemplateDAO.hs`. 
2. In `Main.hs` I will put these `TODO` and ask to resolve them in `README`.

Reply:
Or you can move `repoSettings` into `TemplateDAO.hs`, so that only that file will have to be updated.

Btw, I think we should also consider the case when there is no suitable repository link to put here and make it clearer what to do in that case.

Reply:
> Btw, I think we should also consider the case when there is no suitable repository link to put here and make it clearer what to do in that case.

I also thought about this, but I don't know what to do in this case myself ))

Will try to find a way to provide no git rev.

Reply:
One option is to hardcode `DGitRevisionUnknown` and say that git revision is for advanced users who can tweak it themselves if they want.

Reply:
I was afraid that it will trigger some problemes on autodoc side, but gonna check.

Reply:
Ah wait, a specially added the `DGitRevisionUnknown` for case when there is no related git repository, everything should be fine :facepalm: